backend/app/adapter/routing/handle/search.go

```diff
@@ -1,17 +1,118 @@
 package handle
 
 import (
+	"encoding/json"
+	"io/ioutil"
 	"net/http"
 
 	"github.com/short-d/app/fw/router"
+	"github.com/short-d/short/backend/app/entity"
+	"github.com/short-d/short/backend/app/usecase/authenticator"
 	"github.com/short-d/short/backend/app/usecase/search"
+	"github.com/short-d/short/backend/app/usecase/search/order"
 )
 
-// Search fetches resources under certain criterias.
+// SearchRequest represents the request received from Search API.
+type SearchRequest struct {
+	query  search.Query
+	filter search.Filter
+}
+
+// Search fetches resources under certain criteria.
 func Search(
-	search search.Search,
+	searcher search.Search,
+	authenticator authenticator.Authenticator,
 ) router.Handle {
 	return func(w http.ResponseWriter, r *http.Request, params router.Params) {
-		w.Write([]byte("not implemented"))
+		user := getUser(r, authenticator)
+		if user == nil {
+			w.Write([]byte("user not authenticated"))
+			return
+		}
+
+		searchBody, err := parseSearchBody(r)
+		if err != nil {
+			http.Error(w, err.Error(), 500)
+			return
+		}
+		searchBody.setUser(user)
+
+		w.Write([]byte("request is read"))
+	}
+}
+
+// UnmarshalJSON implements the json.Unmarshaler interface.
+func (s *SearchRequest) UnmarshalJSON(data []byte) error {
+	temp := struct {
+		Query  string `json:"query"`
+		Filter struct {
+			MaxResults int      `json:"max_results"`
+			Resources  []string `json:"resources"`
+			Orders     []string `json:"orders"`
+		} `json:"filter"`
```

Comment:
Let's create a separate `Unmarshal` for Filter. We can specify `resources` and `orders` as enums define in the use case because we are going to convert the type within `Filter`'s `Unmarshal` function.